src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java

```diff
@@ -291,4 +295,30 @@ private boolean isSubmitAction(List<SubmitAction> submitActions) {
     private boolean shouldExport(Set<SubmitAction> submitActions) {
         return submitActions.contains(SubmitAction.EXPORT) || submitActions.contains(SubmitAction.EXPORT_METADATA);
     }
+
+    private void removeGraphValidationErrors(SubmissionEnvelope submissionEnvelope) {
+        biomaterialRepository.findBySubmissionEnvelope(submissionEnvelope)
+                .forEach(biomaterial -> {
+                    biomaterial.setGraphValidationErrors(new ArrayList<>());
+                    biomaterialRepository.save(biomaterial);
+                });
+
+        processRepository.findBySubmissionEnvelope(submissionEnvelope)
+                .forEach(process -> {
+                    process.setGraphValidationErrors(new ArrayList<>());
+                    processRepository.save(process);
+                });
+
+        protocolRepository.findBySubmissionEnvelope(submissionEnvelope)
+                .forEach(protocol -> {
+                    protocol.setGraphValidationErrors(new ArrayList<>());
+                    protocolRepository.save(protocol);
+                });
+
+        fileRepository.findBySubmissionEnvelope(submissionEnvelope)
+                .forEach(file -> {
+                    file.setGraphValidationErrors(new ArrayList<>());
+                    fileRepository.save(file);
+                });
+    }
 }
```

Comment:
could optimise by doing an update with condition

Reply:
Can you explain a bit more?

Reply:
I thought you can do update many using the mongo repository object. Looks like it's not supported. https://github.com/spring-projects/spring-data-mongodb/blob/e28bede416e4ddac19a35dc239388afc90b9cac4/src/main/asciidoc/reference/mongo-repositories.adoc

You could also use saveAll. There's a slight performance improvement. Found this:
https://www.baeldung.com/spring-data-save-saveall

Reply:
Okay I used saveAll